R/epidist_utils.R

```diff
@@ -0,0 +1,605 @@
+#' A helper function when creating uncertainty for the parameters of the
+#' distribution for the epidist object
+#'
+#' @param ci A numeric vector of length two with the lower and upper bound of
+#' the confidence interval or credible interval
+#' @param ci_interval A numeric specifying the interval for the ci, e.g. 95 is
+#' 95% ci
+#' @param ci_type A character string, either "confidence interval" or "credible
+#' interval"
+#'
+#' @return List of three elements
+#' @export
+#'
+#' @examples
+#' # example with uncertainty
+#' create_epidist_uncertainty(
+#'   ci = c(1, 3),
+#'   ci_interval = 95,
+#'   ci_type = "confidence interval"
+#' )
+#' # example with unknown uncertainty
+#' create_epidist_uncertainty(NA)
+create_epidist_uncertainty <- function(ci = NA_real_, ci_interval, ci_type) {
+
+  # when no uncertainty is given
+  if (anyNA(ci)) return(list(
+    ci = NA_real_,
+    ci_interval = c(NA_real_, NA_real_),
+    ci_type = NA_character_
+  ))
+
+  # check input
+  checkmate::assert_numeric(ci, any.missing = FALSE, len = 2)
+  checkmate::assert_number(ci_interval, lower = 0, upper = 100)
+  checkmate::assert_character(ci_type)
+  stopifnot(
+    "ci_type must be either 'confidence interval or credible interval" =
+      all(ci_type %in% c("confidence interval", "credible interval"))
+  )
+
+  # return list of parameter uncertainty
+  list(
+    ci = ci,
+    ci_interval = ci_interval,
+    ci_type = ci_type
+  )
+}
```

Comment:
1. Suggest renaming list elements to "ci" for the actual interval (e.g 95%, currently called "ci_interval"), renaming "ci_interval" (confidence interval interval?) to "ci_limits", a two element vector for the lower and upper limits (similar to `range()` order, smaller first); "ci_type" seems fine.
2. Suggest adding example for creating uncertainty around two parameters, as a number of distributions have two parameters - it's unclear how this should be done, or whether `epidist` objects only support uncertainty in one parameter.
3. Example for unkown uncertainty has superfluous `NA`, and does not match usage in `epidist` documentation examples - suggest removing `NA`.
4. Suggest renaming function title to "Specify distribution parameter uncertainty", current title is too long and is identical to the description.

Reply:
Naming of CI elements have been changed in line with suggestions, resolved in ad66720. New examples have been added to `create_epidist_uncertainty()`, including for multiple parameters and examples for the NA and empty function call cases when uncertainty is unknown.